plugin_tests/segmentation_label_test.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+from tests import base
+
+import numpy as np
+import os
+
+from histomicstk.segmentation.label import trace_boundary
+
+
+# boiler plate to start and stop the server if needed
+def setUpModule():
+    base.enabledPlugins.append('HistomicsTK')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+TEST_DATA_DIR = os.path.join(os.environ['GIRDER_TEST_DATA_PREFIX'],
+                             'plugins/HistomicsTK')
+
+
+class TraceBoundsTest(base.TestCase):
+
+    def test_trace_boundary(self):
+
+        # refenece left neighbor
+        rx = [1, 1, 1]
+        ry = [2, 1, 2]
+
+        # test left neighbor
+        m_left_neighbor = np.array([[0, 0, 0, 0],
+                                    [0, 1, 1, 0],
+                                    [0, 0, 0, 0],
+                                    [0, 0, 0, 0]], dtype=np.bool)
```

Comment:
Since you use `DX = 1, DY = 0` as initial direction in your tracing algorithm, this would be front neighbor i think?